web/src/components/tasks/LicenseTask.tsx

```diff
@@ -77,25 +84,23 @@ export const LicenseTask = (props: Props): ReactElement => {
 
     setIsLoading(true);
     api
-      .checkLicenseStatus(nameAndAddress)
+      .checkLicenseStatus(nameAndAddress, props.task.id as LicenseTaskID)
       .then((result: UserData) => {
-        analytics.event.task_address_form.response.success_application_found();
-        const resultLicenseData = result.businesses[result.currentBusinessId].licenseData;
+        const resultLicenseData =
+          result.businesses[result.currentBusinessId]?.licenseData?.licenses?.[licenseNameForTask];
+
         if (!resultLicenseData) return;
-        setLicenseStatusResult({
-          status: resultLicenseData.status,
-          checklistItems: resultLicenseData.items,
-        });
-        setError(undefined);
-      })
-      .catch((error_) => {
-        if (error_ === 404) {
+        analytics.event.task_address_form.response.success_application_found();
+
+        setLicenseDetails(resultLicenseData);
+        if (resultLicenseData.licenseStatus === "UNKNOWN") {
           analytics.event.task_address_form.response.fail_application_not_found();
           setError("NOT_FOUND");
-        } else {
-          setError("SEARCH_FAILED");
         }
       })
+      .catch(() => {
+        setError("SEARCH_FAILED");
```

Comment:
We are getting this error when no results are returned, should there be a future ticket to address this? Should we address it here?

Reply:
New ticket to address this

Reply:
look into 

         if (webserviceHasInvalidMatch || rgbHasInvalidMatch) {
            return update(userData, {
              nameAndAddress: nameAndAddress,
              licenseStatusResult: {},
              taskIdWithError: taskId,
            });
          }

Reply:
Product need to update the error message

Reply:
created https://www.pivotaltracker.com/story/show/188109500